src/ast/passes/semantic_analyser.cpp

```diff
@@ -216,6 +213,300 @@ class SemanticAnalyser : public Visitor<SemanticAnalyser> {
 
 } // namespace
 
+// clang-format off
+enum class ReturnKind {
+  MAP,
+  VAR,
+  NONE,
+};
+
+struct arg_type_spec {
+  Type type = Type::integer;
+  bool literal = false;
+  // This indicates that this is just a placeholder
+  // as we use the index in the vector of arg_type_spec
+  // as the number argument to check
+  bool skip_check = false;
+};
+
+struct call_spec {
```

Comment:
Can we reconcile this with the `FunctionRegistry`?

Reply:
@amscanne Are we still planning on keeping the FunctionRegistry?